kobo/log.py

```diff
@@ -76,21 +77,25 @@ def add_file_logger(logger, logfile, log_level=None, format=None, mode="a"):
     log_level = log_level or logging.DEBUG
     format = format or BRIEF_LOG_FORMAT
 
+    # Create parent directory if needed.
+    try:
+        os.makedirs(os.path.dirname(logfile))
+    except OSError as exc:
+        if exc.errno != errno.EEXIST:
+            raise
+
     # touch the logfile
     if not os.path.exists(logfile):
-        try:
-            fo = open(logfile, "w")
-            fo.close()
-        except (ValueError, IOError):
-            return
+        fo = open(logfile, "w")
+        fo.close()
 
     # is the logfile really a file?
     if not os.path.isfile(logfile):
```

Comment:
Good point. I didn't consider the `/dev/null` case. I like the idea with warnings: it means this can be still ignored by default, but at least logged on stderr, and it could also be turned to exceptions quite easily. I'll update.